cpu/msp430_common/cpu.c

```diff
@@ -117,19 +117,20 @@ char *thread_stack_init(thread_task_func_t task_func, void *arg, void *stack_sta
     --stackptr;
 
     /* initial value for SR */
-
     *stackptr = GIE;
     --stackptr;
 
-    /* set argument to task_func */
-    *stackptr = (unsigned short) arg;
-    --stackptr;
-
     /* Space for registers. */
-    for (unsigned int i = 14; i > 4; i--) {
+    for (unsigned int i = 15; i > 4; i--) {
         *stackptr = i;
         --stackptr;
     }
```

Comment:
Why not just `stackptr -= 12;`?

Reply:
so the initial values of the register equals the register number. it was like this before, but proved to be a nice hint when manually figuring out the ABI change...